lib/web_ui/flutter_js/src/canvaskit_loader.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import { createWasmInstantiator } from "./instantiate_wasm.js";
+
+export const loadCanvasKit = (deps, config, browserEnvironment, engineRevision) => {
+  if (window.flutterCanvasKit) {
+    // The user has set this global variable ahead of time, so we just return that.
+    return Promise.resolve(window.flutterCanvasKit);
+  }
+  window.flutterCanvasKitLoaded = new Promise((resolve, reject) => {
+    const supportsChromiumCanvasKit = browserEnvironment.hasChromiumBreakIterators && browserEnvironment.hasImageCodecs;
+    if (!supportsChromiumCanvasKit && config.canvasKitVariant == "chromium") {
+      throw "Chromium CanvasKit variant specifically requested, but unsupported in this browser";
+    }
+    const useChromiumCanvasKit = supportsChromiumCanvasKit && (config.canvasKitVariant !== "full");
+    let baseUrl = config.canvasKitBaseUrl ?? `https://www.gstatic.com/flutter-canvaskit/${engineRevision}/`;
+    if (useChromiumCanvasKit) {
+      baseUrl = `${baseUrl}/chromium/`;
+    }
+    let canvasKitUrl = `${baseUrl}canvaskit.js`;
+    if (deps.flutterTT.policy) {
```

Comment:
QQ: how do we decide what goes in `deps` vs what is passed as a parameter to the function?

For example, the `browserEnvironment` looks like something that could be a `dep` too, and not need its own specific positional parameter in the `loadCanvasKit` function?

Reply:
I guess it's mostly arbitrary. The browser environment just logically felt like something different than a "dependency" so I have it as a separate thing. Perhaps I don't even need the `browserEnvironment` to be passed as a parameter anymore now that it's just a global variable vended by the `browser_environment.js` module...

Reply:
> now that it's just a global variable vended by the browser_environment.js module

I thought the same, but also if you grab it from deps (or the parameter) it's easy(er) to mock for testing maybe?